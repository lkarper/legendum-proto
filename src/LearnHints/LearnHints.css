.LearnHints__hide-show-button {
    margin: 5px;
}