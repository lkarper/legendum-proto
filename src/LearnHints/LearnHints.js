import React from 'react';

const LearnHints = (props) => {
    const { page, showHintsBoolean, setShowHintsBoolean } = props;

    const hideAndShowButton = showHintsBoolean 
            ? <button onClick={() => setShowHintsBoolean(false)}>Hide hints</button>
            : <button onClick={() => setShowHintsBoolean(true)}>Show hints</button>
        ;
    
    const hintsList = (
        <ol>
            {page.hints
                .split('|')
                .map((hint, i) => (
                    <li key={i}>
                        <p>{hint}</p>
                        <button>Save hint</button>
                    </li>
                ))}
        </ol>
    );

    return (
        <div>
            {hideAndShowButton}
            {showHintsBoolean ? hintsList : ''}
        </div>
    );
}

export default LearnHints;