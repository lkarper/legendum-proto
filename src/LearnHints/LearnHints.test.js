import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LearnHints from './LearnHints';

describe('LearnHints component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<LearnHints />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<LearnHints />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
