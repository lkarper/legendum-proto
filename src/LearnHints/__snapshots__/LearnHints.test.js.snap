// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnHints component renders the UI as expected 1`] = `
<LearnHints
  page={
    Object {
      "hints": Array [],
    }
  }
  setShowHintsBoolean={[Function]}
  showHintsBoolean={false}
>
  <div
    className="LearnHints__container"
  >
    <button
      className="LearnHints__hide-show-button button"
      onClick={[Function]}
    >
      Show hints
    </button>
  </div>
</LearnHints>
`;
